### PR TITLE
upgrade kubebuilder envtest to v3

### PR DESCRIPTION
**Pull request by yuswift**
Signed-off-by: yuswift <yuswift2018@gmail.com>

### What type of PR is this?
/kind bug
/kind failing-test

### What this PR does / why we need it:
We used to use `hack/install_kubebuilder.sh` to install `kubebuillder` libraries(e.g. etcd, kube-apiserver). However, the libraries version is 1.16, which is not compatible with 1.22. I updated the Makefile by referring to the [docs](https://book.kubebuilder.io/migration/manually_migration_guide_v2_v3.html?highlight=envtes#to-allow-automatic-downloads).
BTW, the `hack/install_kubebuilder.sh` file is no longer needed, we can remove it in the future. I've opened a new issue.
### Which issue(s) this PR fixes:

part of #4127 

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?
```release-note
Bump kubebuilder envtest to v3, the users don't need to install the kubebuilder with `hack/install_kubebuilder.sh` anymore.
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -137,8 +144,11 @@ helm-uninstall: ; $(info $(M)...Begin to helm-uninstall.)  @ ## Helm-uninstall.
 	kubectl delete -f https://raw.githubusercontent.com/kubesphere/ks-installer/master/roles/ks-core/prepare/files/ks-init/role-templates.yaml
 
 # Run tests
+ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
 test: vet ;$(info $(M)...Begin to run tests.)  @ ## Run tests.
-	export KUBEBUILDER_CONTROLPLANE_START_TIMEOUT=2m; go test ./pkg/... ./cmd/... -covermode=atomic -coverprofile=coverage.txt
+	mkdir -p ${ENVTEST_ASSETS_DIR}
+	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.8.3/hack/setup-envtest.sh
```
Comment (zryfish) [inline Makefile:150]: I am thinking could we add a new entry, for example `dev-env` to prepare a local development environment, and download all necessary resources.

Comment (yuswift) [reply Makefile:150]: What do you mean `all necessary resources`? All I know are `kubebuilder` related resources. The `dev-env` dir name looks good to me.

Comment (zryfish) [reply Makefile:150]: I mean a new make entry,  like https://github.com/kubernetes/ingress-nginx/blob/main/Makefile#L151

Comment (yuswift) [reply Makefile]: done

**Pull request merged**